FHEM/00_SIGNALduino.pm

```diff
@@ -142,53 +142,82 @@ my @syncmod = ( 'No preamble/sync','15/16 sync word bits detected','16/16 sync w
-  '05' => 'SYNC0   ',
-  '06' => 'PKTLEN  ',
-  '07' => 'PKTCTRL1',
-  '08' => 'PKTCTRL0',
-  '09' => 'ADDR    ',
-  '0A' => 'CHANNR  ',
-  '0B' => 'FSCTRL1 ',
-  '0C' => 'FSCTRL0 ',
-  '0D' => 'FREQ2   ',
-  '0E' => 'FREQ1   ',
-  '0F' => 'FREQ0   ',
-  '10' => 'MDMCFG4 ',
-  '11' => 'MDMCFG3 ',
-  '12' => 'MDMCFG2 ',
-  '13' => 'MDMCFG1 ',
-  '14' => 'MDMCFG0 ',
-  '15' => 'DEVIATN ',
-  '16' => 'MCSM2   ',
-  '17' => 'MCSM1   ',
-  '18' => 'MCSM0   ',
-  '19' => 'FOCCFG  ',
-  '1A' => 'BSCFG   ',
-  '1B' => 'AGCCTRL2',
-  '1C' => 'AGCCTRL1',
-  '1D' => 'AGCCTRL0',
-  '1E' => 'WOREVT1 ',
-  '1F' => 'WOREVT0 ',
-  '20' => 'WORCTRL ',
-  '21' => 'FREND1  ',
-  '22' => 'FREND0  ',
-  '23' => 'FSCAL3  ',
-  '24' => 'FSCAL2  ',
-  '25' => 'FSCAL1  ',
-  '26' => 'FSCAL0  ',
-  '27' => 'RCCTRL1 ',
-  '28' => 'RCCTRL0 ',
-  '29' => 'FSTEST  ',
-  '2A' => 'PTEST   ',
-  '2B' => 'AGCTEST ',
-  '2C' => 'TEST2   ',
-  '2D' => 'TEST1   ',
-  '2E' => 'TEST0   ',
+  '00' => 'IOCFG2   - 0x0D',			# ! the values with spaces for output get ccreg 99 !
+  '01' => 'IOCFG1   - 0x2E',
+  '02' => 'IOCFG0   - 0x2D',
+  '03' => 'FIFOTHR  - 0x47',
+  '04' => 'SYNC1    - 0xD3',
+  '05' => 'SYNC0    - 0x91',
+  '06' => 'PKTLEN   - 0x3D',
+  '07' => 'PKTCTRL1 - 0x04',
+  '08' => 'PKTCTRL0 - 0x32',
+  '09' => 'ADDR     - 0x00',
+  '0A' => 'CHANNR   - 0x00',
+  '0B' => 'FSCTRL1  - 0x06',
+  '0C' => 'FSCTRL0  - 0x00',
+  '0D' => 'FREQ2    - 0x10',
+  '0E' => 'FREQ1    - 0xB0',
+  '0F' => 'FREQ0    - 0x71',
+  '10' => 'MDMCFG4  - 0x57',
+  '11' => 'MDMCFG3  - 0xC4',
+  '12' => 'MDMCFG2  - 0x30',
+  '13' => 'MDMCFG1  - 0x23',
+  '14' => 'MDMCFG0  - 0xB9',
+  '15' => 'DEVIATN  - 0x00',
+  '16' => 'MCSM2    - 0x07',
+  '17' => 'MCSM1    - 0x00',
+  '18' => 'MCSM0    - 0x18',
+  '19' => 'FOCCFG   - 0x14',
+  '1A' => 'BSCFG    - 0x6C',
+  '1B' => 'AGCCTRL2 - 0x07',
+  '1C' => 'AGCCTRL1 - 0x00',
+  '1D' => 'AGCCTRL0 - 0x91',
+  '1E' => 'WOREVT1  - 0x87',
+  '1F' => 'WOREVT0  - 0x6B',
+  '20' => 'WORCTRL  - 0xF8',
+  '21' => 'FREND1   - 0xB6',
+  '22' => 'FREND0   - 0x11',
+  '23' => 'FSCAL3   - 0xE9',
+  '24' => 'FSCAL2   - 0x2A',
+  '25' => 'FSCAL1   - 0x00',
+  '26' => 'FSCAL0   - 0x1F',
+  '27' => 'RCCTRL1  - 0x41',
+  '28' => 'RCCTRL0  - 0x00',
+  '29' => 'FSTEST   - N/A ',
+  '2A' => 'PTEST    - N/A ',
+  '2B' => 'AGCTEST  - N/A ',
+  '2C' => 'TEST2    - N/A ',
+  '2D' => 'TEST1    - N/A ',
+  '2E' => 'TEST0    - N/A ',
+);
+
+my %cc1101_status_register = ( # for get ccreg 30-3D status registers
+  '30' => 'PARTNUM       ',
+  '31' => 'VERSION       ',
+  '32' => 'FREQEST       ',
+  '33' => 'LQI           ',
+  '34' => 'RSSI          ',
+  '35' => 'MARCSTATE     ',
+  '36' => 'WORTIME1      ',
+  '37' => 'WORTIME0      ',
+  '38' => 'PKTSTATUS     ',
+  '39' => 'VCO_VC_DAC    ',
+  '3A' => 'TXBYTES       ',
+  '3B' => 'RXBYTES       ',
+  '3C' => 'RCCTRL1_STATUS',
+  '3D' => 'RCCTRL0_STATUS',
+);
+
+my %cc1101_version = ( # Status register 0x31 (0xF1): VERSION – Chip ID
```

Comment:
Ich würde annehmen, dass der Teil nicht ins package main gehört, wenn dann eher in package cc1101 oder?

Prinzipiell hatte ich an anderer Stelle schon geäußert, dass diese Zuordnung nicht mit dem SIGNALduino zu tun hat, und eine lib für den cc1101 geeigneter / wiederverwendbar durch andere Module wäre.

Reply:
Meinst du damit, diese Liste innerhalb der 00_SIGNALduino.pm zu verschieben in package cc1101, oder gleich ein neues Modul anlegen ähnlich wie SD_Protocols.pm?

Reply:
Mindestens mal in das package ein eigenes Perl Modul wäre besser für die mitnutzung, aber dann müssten noch ein paar Dinge mehr verschoben werden, also vielleicht erst mal kleine Brötchen backen.

Reply:
Ich bleibe bei den kleinen Brötchen :-)